### DISCUSSION TITLE
How load dynamic class

### DISCUSSION THREAD

**Discussion by l1mindr** [tailwindlabs/tailwindcss#10361] [Help] [answered]
Hi
I'm using tailwind on nextjs 
this is working on development 
```css
className={`text-${color}-500 bg-${color}-100`}
```
but on production not working 
‌‌how can I fix it?

**Comment by wongjn** [comment, marked as answer]
[From the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> If you’re using a component library like React or Vue, this means you shouldn’t use props to dynamically construct classes:
> 
> ❌ Don't use props to build class names dynamically
> 
> ```jsx
> function Button({ color, children }) {
>   return (
>     <button className={`bg-${color}-600 hover:bg-${color}-500 ...`}>
>       {children}
>     </button>
>   )
> }
> ```
> Instead, map props to complete class names that are statically detectable at build-time:
> 
> ✔ Always map props to static class names
> 
> ```jsx
> function Button({ color, children }) {
>   const colorVariants = {
>     blue: 'bg-blue-600 hover:bg-blue-500',
>     red: 'bg-red-600 hover:bg-red-500',
>   }
> 
>   return (
>     <button className={`${colorVariants[color]} ...`}>
>       {children}
>     </button>
>   )
> }
> ```
